package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -0,0 +1,335 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+"""
+Reading trajectory with Chemfiles --- :mod:`MDAnalysis.coordinates.chemfiles`
+=============================================================================
+
+Classes to read and write files using the `chemfiles`_ library. This library
+provides C++ implementation of multiple formats readers and writers, the full
+list if available `here <formats>`_.
+
+.. _chemfiles: https://chemfiles.org
+.. _formats: http://chemfiles.org/chemfiles/latest/formats.html
+
+Classes
+-------
+
+.. autoclass:: ChemfilesReader
+
+.. autoclass:: ChemfilesWriter
+
+"""
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+
+import numpy as np
+from six import string_types
+
+from . import base, core
+
+import chemfiles
+from chemfiles import Trajectory, Frame, Atom, UnitCell, Residue, Topology
+
+
+assert chemfiles.__version__.startswith("0.9.")
```

Comment:
I would move the assert into the `__init__` function so that the import of mdanalysis doesn't crash the interpreter if the version is to old. This way one can still use the library when the chemfiles reader is not used. In python, I would also use a RuntimeError exception instead of an assert, this way you can add an easy to read error message.

```python

def check_chemfiles_version(min_version_name='0.9'):
    from distutils.version import LooseVersion
    chemfiles_version = LooseVersion(chemfiles.__version__)
    min_version = LooseVersion(min_version_name)
    if chemfiles_version < min_version:
        raise RuntimeError("minimum required Chemfiles version is {}, found: {}".format(
            min_version_name, chemfiles.__version__)

```

This parsing will also be a bit more robust. The function can be added as is to `__init__`.

Reply:
Good idea!